File: src/javagameengine/components/Camera.java
package javagameengine.components;

import javagameengine.JavaGameEngine;
import javagameengine.msc.Vector2;

public class Camera extends Component{


    @Override
    public void start() {
        super.start();
        setPosition(new Vector2(JavaGameEngine.getWindowSize().getX()/2,JavaGameEngine.getWindowSize().getY()/2));
    }

    @Override
    public Vector2 getScale() {
        return scale;
    }

    @Override
    public void setScale(Vector2 scale) {
        this.scale = scale;
    }
}

File: src/javagameengine/components/Camera.kt
package javagameengine.components

import javagameengine.msc.Vector2

class Camera : Component() {

    init {
        tag = "Camera"
        scale = Vector2(1f,1f);
        position = Vector2(0f,0f);
    }

    override fun setPosition(position: Vector2?) {
        this.position = position
    }

    override fun getScale(): Vector2? {
        return scale
    }

}